src/TwilioVideo.android.js

```diff
@@ -309,6 +320,16 @@ class CustomTwilioVideoView extends Component {
       "onLocalParticipantSupportedCodecs",
     ].reduce((wrappedEvents, eventName) => {
       if (this.props[eventName]) {
+        if (eventName === "onDataTrackBinaryMessageReceived") {
+          return {
+            ...wrappedEvents,
+            [eventName]: (data) =>
+              this.props[eventName]({
+                message: toByteArray(data.nativeEvent.message),
```

Comment:
🛑 You're dropping the other parts of `nativeEvent` (i.e. the track sid). You need to preserve all other parts of `nativeEvent` and just replace the value of `message`.